test/e2e/page-objects/pages/test-dapp-solana.ts

```diff
@@ -0,0 +1,365 @@
+import { text } from 'stream/consumers';
+import { WINDOW_TITLES } from '../../helpers';
+import { Driver } from '../../webdriver/driver';
+import { dataTestIds } from '@metamask/test-dapp-solana';
+import { By } from 'selenium-webdriver';
+const DAPP_HOST_ADDRESS = '127.0.0.1:8080';
+const DAPP_URL = `http://${DAPP_HOST_ADDRESS}`;
+
+export class TestDappSolana {
+  constructor(private readonly driver: Driver) {}
+
+  /**
+   * Open the solana test dapp page.
+   *
+   * @param options - The options for opening the test dapp page.
+   * @param options.url - The URL of the dapp. Defaults to DAPP_URL.
+   * @returns A promise that resolves when the new page is opened.
+   */
+  async openTestDappPage({
+    url = DAPP_URL,
+  }: {
+    url?: string;
+  } = {}): Promise<void> {
+    await this.driver.openNewPage(url);
+  }
+
+  /**
+   * Focus on the Solana test dapp window.
+   */
+  async switchTo() {
+    await this.driver.switchToWindowWithTitle(WINDOW_TITLES.SolanaTestDApp);
+  }
+
+  /**
+   * Get wallet modal.
+   *
+   * @returns The wallet modal component helper methods.
+   */
+  async getWalletModal() {
+    await this.driver.waitForSelector('.wallet-adapter-modal-list');
+
+    const walletButtons = await this.driver.findElements(
+      '.wallet-adapter-modal-list .wallet-adapter-button ',
```

Comment:
should this selector be moved into a constant on this class? seems like the pattern other page objects follow